neard/tests/node_cluster.rs

```diff
@@ -0,0 +1,106 @@
+use futures::future;
+
+use near_actix_test_utils::{run_actix_until_stop, spawn_interruptible as spawn};
+use near_client::{ClientActor, ViewClientActor};
+use near_primitives::types::{BlockHeight, BlockHeightDelta, NumSeats, NumShards};
+use testlib::{start_nodes, test_helpers::heavy_test};
+
+pub enum ClusterConfigVariant {
+    HeavyTest(bool),
+    Shards(NumShards),
+    ValidatorSeats(NumSeats),
+    LightClients(usize),
+    EpochLength(BlockHeightDelta),
+    GenesisHeight(BlockHeight),
+}
+
+use ClusterConfigVariant::*;
+
+#[derive(Default, Debug)]
+pub struct NodeCluster {
+    dirs: Vec<tempfile::TempDir>,
+    is_heavy: bool,
+    num_shards: Option<NumShards>,
+    num_validator_seats: Option<NumSeats>,
+    num_lightclient: Option<usize>,
+    epoch_length: Option<BlockHeightDelta>,
+    genesis_height: Option<BlockHeight>,
+}
+
+impl NodeCluster {
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    pub fn with(mut self, config: ClusterConfigVariant) -> Self {
+        match config {
+            HeavyTest(is_heavy) => self.is_heavy = is_heavy,
+            Shards(n) => self.num_shards = Some(n),
+            ValidatorSeats(n) => self.num_validator_seats = Some(n),
+            LightClients(n) => self.num_lightclient = Some(n),
+            EpochLength(l) => self.epoch_length = Some(l),
+            GenesisHeight(h) => self.genesis_height = Some(h),
+        };
+        self
```

Comment:
As an alternative, I think we can avoid defining `ClusterConfigVariant` struct and just create a bunch of `with` functions. Either variant works, but I'd personally go with a bunch of functions, as that feels less fancy and doesn't require an extra import.

Reply:
@frol, what do you think?

<table>
<tr>
<td>

``` rust
cluster
    .with(HeavyTest(true))
    .with(Shards(1))
    .with(ValidatorSeats(1))
    .with(LightClients(0))
    .with(EpochLength(10))
    .with(GenesisHeight(0));
```
</td>
<td>

``` rust
cluster
    .heavy_test(true)
    .shards(1)
    .validator_seats(1)
    .light_clients(1)
    .epoch_length(10)
    .genesis_height(0);
```
</td>
</tr>
<tr>
<td> Current </td> <td> Proposed </td>
</tr>
</table>

Or are they to be prefixed with `set_` e.g `.set_genesis_height(0)`

Reply:
Yeah, I would go with the proposed alt variant. `is_heavy_test(true)` could be just `set_heavy_test()` (don't need to even specify the argument)